Added option to specify the continued flag in the command line

rerodrigues opened the pull request:
I've added an option to set the `continued` flag of a grunt-multi task directly from command line

#### Usage

```bash
#defaults to true
grunt multi:any_task --continued

#switches to true or false
grunt multi:any_task --continued=true|false
```

This was created with the changes of the PR #5 in mind, but can also be used independently to turn on/off the `continued` flag in any grunt-multi task.

obs.: This PR also includes the typo fixes from PR #4, but NOT the changes from PR #5!

neekey commented:
hi @rerodrigues, thanks for the contribution, I understand the user case here, but I think it will be better we can handle all the option related flag together and make them distinguishable with the "vars" flags, bellow is my proposal:

```bash
grunt multi:any_task page_list=a,b,c --option-continued=true|false
grunt multi:any_task page_list=a,b,c --option-maxSpawn=10
grunt multi:any_task page_list=a,b,c --option-tasks=10
```

We can only support these three options `continued`, `maxSpawn` and `tasks`, and all the options set via flags will overwrite task or target options.

Same opinion applies to https://github.com/neekey/grunt-multi/pull/5

rerodrigues commented:
Hi @neekey thanks for your reply.

Agreed! It makes sense to keep things separated in a more organized way. I'll update the PRs to reflect your proposal and include the maxSpawn flag as well.

One more thing, I'm in doubt between `--option-max-spawn` and `--option-maxSpawn`. `--option-max-spawn` is more standardized than `--option-maxSpawn`, but I think it looks a bit odd. What do you think?

neekey commented:
Good suggestion, I think --option-max-spawn will be better : )

On Tuesday, 8 November 2016, Renato Rodrigues <notifications@github.com>
wrote:

> Hi @neekey <https://github.com/neekey> thanks for your reply.
>
> Agreed! It makes sense to keep things separated in a more organized way.
> I'll update the PRs to reflect your proposal and include the maxSpawn flag
> as well.
>
> One more thing, I'm in doubt between --option-max-spawn and
> --option-maxSpawn. --option-max-spawn is more standardized than
> --option-maxSpawn, but I think it looks a bit odd. What do you think?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/neekey/grunt-multi/pull/6#issuecomment-258922162>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AAegnr03_Yn0XeDqDiQ2EyJpT8cxURlcks5q72_rgaJpZM4KqW4K>
> .
>


-- 
发自移动版 Gmail

rerodrigues commented:
Changed the --multi-tasks flag to --option-tasks to comply with the -option-xxx proposal.

I've tried to concentrate all the option flags parsing in one place, inside the new function created in the PR #5, but it was impossible due the way grunt.option.flags() return flags:

grunt.option.flags() returns:

```
--option-continued --> --option-continued
--option-continued=true --> --option-continued
--option-continued=something --> --option-continued=something
```
and the problematic part
```
--option-continued=false --> **nothing**
```
I cannot specify --option-continued=false and read it using grunt.option.flags(), since it will make the xxx=false, flag to vanish and I would never know they existed. For this reason I implemented this handler outside of the options parsing function.

neekey commented:
Hi @rerodrigues! Sorry for away for a long time. I was busy with some personal stuff after I got out of the hospital.

Glad to see you have some update. I'm thinking about that you can merge PR5 into this PR since they kind of go into the same direction `--option-xx` and you don't need to sync these two PRs anymore.

Anyway, it looks like the only problem left is the `---option-xxx=false` not caught by `grunt.option.flags()` and I'm thinking maybe we can just treat it as false if it is not provided or set to `false` explicitly.

rerodrigues commented:
Hi @neekey! Glad to hear that you are better!

As suggested, I've merged this PR with PR5. Now all the three --option-xxx functionalities are covered in this PR.

About the `--option-continued`, I handled it outside of the `.flags()` loop, manually checking for `grunt.option('option-continued')` and it's working like a charm, even for `--option-continued` (without `true` or `false` specified)
